src/universal/containers/DashLayoutContainer/DashLayoutContainer.js

```diff
@@ -81,15 +83,28 @@ export default class DashLayoutContainer extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
+    const {activeMeetings, trialNotification} = this.props;
+    const {barType} = trialNotification;
     if (this.props.tms !== nextProps.tms) {
       subToAllTeams(nextProps.tms);
     }
+
+    if (activeMeetings.length > 0 || barType === TRIAL_EXPIRES_SOON || barType === TRIAL_EXPIRED) {
+      this.props.dispatch(notificationBarPresent(true));
+    } else {
+      this.props.dispatch(notificationBarPresent(false));
```

Comment:
@ackernaut here's the pattern i use:
- factor this out into its own method
- call that method from componentWillMount and componentWillReceiveProps
- from cWRP, pass in this.props and nextProps
- from cWM, pass in this.props as the nextProps param and {} for nextProps

Check out `MeetingContainer` for the pattern, i do this for `handleRedirects`.